### PR TITLE
Scheduled Updates: Show Jetpack users migrate upsell banner

### PR DESCRIPTION
**Pull request by ouikhuan**
Related to #88655

## Proposed Changes

* If a user is coming from a Jetpack site, we show a migration upsell instead and navigate them to the migration flow.
* Adding upsell banner track events:
    * Simple site:
        - show: `calypso_scheduled_updates_upgrade_impression`
        - click: `calypso_scheduled_updates_upgrade_click`
    * Jetpack site:
        - show: `calypso_scheduled_updates_migrate_impression`
        - click: `calypso_scheduled_updates_migrate_click`
       
## Testing Instructions

* Create a new JN site and connect to your WP.com account.
* Navigate to `http://calypso.localhost:3000/plugins/scheduled-updates/<site-slug>`
* Check and see if the migrate upsell banner shows up
Screen Shot 2024-03-19 at 5 19 02 PM
* Click `Migrate` button and see if it redirects you to the `import-hosted-site` flow
* Check in Tracks Vigilante and see if `calypso_scheduled_updates_migrate_impression` and `calypso_scheduled_updates_migrate_click` both gets triggered.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by ouikhuan** [comment]
Tagging @Automattic/quake team here for review as well to make sure the query string added here https://github.com/Automattic/wp-calypso/pull/88658/files#diff-c5c492c3a551361aa5e0ba5586c126789c1e98051768c57f8adb9e0643726226R52-R54 is correct.

File: client/components/scheduled-updates/scheduled-updates-gate/upsell-nudge.tsx
Comment (obenland) [inline client/components/scheduled-updates/scheduled-updates-gate/upsell-nudge.tsx]: I don't think WoA sites will ever see `UpsellNudgeNotice`, but in case we'll open Atomic to non-Creator plans in the future, can we make sure `isJetpack` does not include WoA sites?

Comment (ouikhuan) [reply client/components/scheduled-updates/scheduled-updates-gate/upsell-nudge.tsx]: Totally, updated!

**Review by bogiii — APPROVED** [review]
Tested, looks good to me! 👍

**Comment by obenland** [comment]
@ouikhuan Anything keeping this from going in?

**Comment by ouikhuan** [comment]
> @ouikhuan Anything keeping this from going in?

I was waiting for @Automattic/quake for review since we redirect them to the import-hosted-site flow, and as far as I know, they use `source` and `ref` to track, so want to make sure nothing is wrong before merging the PR.

**Comment by mpkelly** [comment]
Sorry for the delay. I don't see anything wrong with this PR. I figured `source` was being used in Tracks, but I don't see it. So I guess it must be used for routing logic, although I looked for that too and couldn't see anything specific for a `source` of `sites-dashboard`. 

If your PR is working as expected, I would say merge it. 

CC @zaguiini for a post-deploy review, in case I missed something, as I think Luis last changed this code:

https://github.com/Automattic/wp-calypso/pull/77412/files#diff-340e869ae99622e4bf38c32df4dd55ebb2cad3e3a3ef073da186d4baccb6fd10R3

**Comment by ouikhuan** [comment]
Thanks for the review @mpkelly !

**Comment by ouikhuan** [comment]
Thanks for the feedback @mrfoxtalbot ! I'll take care of this 👍

**Pull request merged**
